frontend/apps/reader/modules/readerfont.lua

```diff
@@ -68,37 +94,31 @@ function ReaderFont:setupFaceMenuTable()
     })
     -- Font list
     cre = require("document/credocument"):engineInit()
-    local face_list = cre.getFontFaces()
+    local face_list = self:setupFaceList(cre.getFontFaces())
     face_list = self:sortFaceList(face_list)
     for k, v in ipairs(face_list) do
+        local lfacename  -- localized facename
         local font_filename, font_faceindex, is_monospace = cre.getFontFaceFilenameAndFaceIndex(v)
-        table.insert(self.face_table, {
-            text_func = function()
-                -- defaults are hardcoded in credocument.lua
-                local default_font = G_reader_settings:readSetting("cre_font") or self.ui.document.default_font
-                local fallback_font = G_reader_settings:readSetting("fallback_font") or self.ui.document.fallback_fonts[1]
-                local monospace_font = G_reader_settings:readSetting("monospace_font") or self.ui.document.monospace_font
-                local text = v
-                if font_filename and font_faceindex then
-                    text = FontList:getLocalizedFontName(font_filename, font_faceindex) or text
-                end
+        if font_filename and font_faceindex then
+            lfacename = FontList:getLocalizedFontName(font_filename, font_faceindex) or v
+        end
 
-                if v == monospace_font then
-                    text = text .. " \u{1F13C}" -- Squared Latin Capital Letter M
-                elseif is_monospace then
-                    text = text .. " \u{1D39}" -- Modified Letter Capital M
-                end
-                if v == default_font then
-                    text = text .. "   ★"
-                end
-                if v == fallback_font then
-                    text = text .. "   �"
-                end
-                if newly_added_fonts[v] then
-                    text = text .. "  \u{EA93}" -- "NEW" in a black square, from nerdfont
-                end
-                return text
-            end,
+        local menu_table
+        local face_menu_text = self:decorateFacename(lfacename, is_monospace)
+        if string.match(v, "^Noto ") then
+            if noto_fonts == nil then
+                table.insert(self.face_table, {
+                    text = _("Noto fonts"),
+                    sub_item_table_func = function() return noto_fonts end,
+                })
+                noto_fonts = {}  -- prep the Noto fonts submenu table
+            end
+            menu_table = noto_fonts
+        else
+            menu_table = self.face_table
+        end
+        table.insert(menu_table, {
+            text_func = function() return face_menu_text end,
```

Comment:
I think this was somehow dynamic (dunno if we rebuilt the menu table when we set a font as default or fallback).
But by building `face_menu_text` outside, even if you use a text_func=, you will return always the same value. (And so, if this is ok, you might as well just use `text = face_menu_text,`

Reply:
This should just be `text = face_menu_text`
Decorating the Facename string (with 'new' or 'M' or whatever) used to happen here, but happens a bit above now.